src/arch/xtensa/CMakeLists.txt

```diff
@@ -61,6 +61,16 @@ else()
 	set(stdlib_flag "-nostdlib")
 endif()
 
+if(CONFIG_OPTIMIZE_FOR_PERFORMANCE)
+	set(optimization_flag "-O2")
+elseif(CONFIG_OPTIMIZE_FOR_SIZE)
+	set(optimization_flag "-Os")
+elseif(CONFIG_OPTIMIZE_FOR_DEBUG)
+	set(optimization_flag "-Og")
+elseif(CONFIG_OPTIMIZE_FOR_NONE)
+	set(optimization_flag "-O0")
+endif()
```

Comment:
maybe add a default value here to avoid the situation about kconfig not updated?

Reply:
Yeah, I agree that an `else() set(optimization_flag "-O2")` would be a good fix for this particular situation. But I believe we should always be careful about introducing Kconfig changes for the other cases when we introduce new configs.

I mean either that or requiring devs to from time to time do a `make oldconfig` (but not very often). I don't think it's worth adding workarounds for this issue unless they're easy like in this case.

Reply:
@paulstelian97 @xiulipan In real world no developer will run `make oldconfig` from time to time ;)

I just need to make generated/.config depend on Kconfig files that trigger make oldconfig and then it will trigger also CMake rerun as it depends on .config

Reply:
Alright, anyway in this particular case it's an easy workaround so it can just be applied immediately.

Reply:
OK, I think no problem here. We have some scripts here.